qiita_db/user.py

```diff
@@ -280,15 +277,34 @@ def verify_code(cls, email, code, code_type):
             return False
 
         db_code = db_code[0]
+        if not db_code:
+            raise QiitaDBError("No %s code for user %s" %
+                               (column, email))
 
         if db_code == code and code_type == "create":
             # verify the user
-            level = conn_handler.execute_fetchone(
-                "SELECT user_level_id FROM qiita.user_level WHERE "
-                "name = %s", ("user", ))[0]
-            sql = ("UPDATE qiita.{} SET user_level_id = %s WHERE "
-                   "email = %s".format(cls._table))
+            level = convert_to_id('user', 'user_level', 'name')
+            sql = """UPDATE qiita.{} SET user_level_id = %s
+                     WHERE email = %s""".format(cls._table)
             conn_handler.execute(sql, (level, email))
+            # create user default sample holders once verified
+            # create one per portal
+            analysis_sql = """INSERT INTO qiita.analysis
+                (email, name, description, dflt, analysis_status_id)
+                VALUES (%s, %s, %s, %s, 1)"""
+            analysis_args = []
+            sql = "SELECT portal_type_id from qiita.portal_type"
+            for portal in conn_handler.execute_fetchall(sql):
+                analysis_args.append(
+                    (email, '%s-dflt-%d' % (email, portal[0]), 'dflt', True))
+            conn_handler.executemany(analysis_sql, analysis_args)
+
+        if db_code == code:
```

Comment:
This comparison is done up to 3 times in the code (previous if, this one and return statement). Can be computed once and stored in a variable?